keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,805 @@
+
+* Monitoring device health or status.
+
+* Performing diagnostics and troubleshooting on devices shared across users.
+
+Enabling such administrative tasks without compromising security is critical to cluster operations. Non-admin users gaining access to devices in use by others could result in:
+
+* Unauthorized access to confidential information about device usage.
+
+* Potential conflicts or misuse of shared hardware.
+
+As the adoption of DRA expands, the lack of privileged administrative access becomes a bottleneck for cluster operations, particularly in shared environments where devices are critical resources.
+
+### Goals
+
+* Provide a privileged mode for cluster administrators to access devices already in use.
+
+* Ensure that only authorized users can use the privileged mode.
+
+* Maintain security and prevent unauthorized access to devices.
+
+* Allow administrators to mark `ResourceClaim` or `ResourceClaimTemplate` objects with an admin access flag in namespaces marked with the DRA admin label.
+
+### Non-Goals
+
+* Redesign DRA.
+
+* Extend privileged mode access to non-admin users.
+
+* Define how administrators use the data accessed through privileged mode.
+
+## Proposal
+
+### DRAAdminAccess Overview
+
+This proposal enhances the `DRAAdminAccess` feature to enable cluster administrators to mark requests in `ResourceClaim` and `ResourceClaimTemplate` objects as privileged. This feature includes:
+
+1. Admin Access Flag:
+
+    A field in `ResourceClaim` and `ResourceClaimTemplate` specifications to indicate the request is in admin mode.
+
+    Example:
+    ```yaml
+    apiVersion: resource.k8s.io/v1beta1
+    kind: ResourceClaim
+    metadata:
+      name: admin-resource-claim
+      namespace: admin-namespace
+    spec:
+      resourceClassName: admin-resource-class
+      adminAccess: true
+    ```
+1. Namespace Label for DRA Admin Mode:
+
+    Namespaces must be labeled to allow the creation of admin-access claims.
+
+    Example label:
+    ```yaml
+    metadata:
+      labels:
+        kubernetes.io/dra-admin-access: "true"
+    ```
+1. Authorization Check:
+
+    In the REST storage layer, validate requests to create `ResourceClaim` or `ResourceClaimTemplate` objects with `adminAccess: true`. Only authorize if namespace has the `kubernetes.io/dra-admin-access` label.
+
+1. Grants privileged access to the requested device:
+
+    For requests with `adminAccess: true`, the DRA controller bypasses standard allocation checks and allows administrators to access devices already in use. This ensures privileged tasks like monitoring or diagnostics can be performed without disrupting existing allocations. The controller also logs and audits admin-access requests for security and traceability.
```

Comment:
Is there an assumption that monitoring/diagnostics processes are already running on the underlying host OS, and so any CPU/memory allocation can be guaranteed? And/or do we assume that any new operational headroom required by these privileged tasks is already pre-accounted for, no chance for container allocation to take up 100% of the available headroom of a node's host OS?

Reply:
This field does not influence CPU or memory consumption. Those monitoring pods must declare their resources and they will get set aside for them when the pods get scheduled.